src/shelly_sw_service.c

```diff
@@ -100,6 +116,25 @@ static void shelly_sw_set_state_ctx(struct shelly_sw_service_ctx *ctx,
       mgos_set_timer(600, 0, do_reset, ctx);
     }
   }
+
+  if (cfg->auto_off && auto_off_timer_id >= 0)
```

Comment:
wrap this whole section in `if (cfg->auto_off)`, this will make other conditions simpler.